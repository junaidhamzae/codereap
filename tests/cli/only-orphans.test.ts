import path from 'node:path';
import fs from 'node:fs';
import os from 'node:os';
import { runCli } from '../helpers/cli';

describe('CLI --onlyOrphans with target enrichment', () => {
  it('includes symbols.imports with target info for resolvable imports', () => {
    const fixture = path.resolve(__dirname, '../fixtures/only-orphans');
    const tmp = fs.mkdtempSync(path.join(os.tmpdir(), 'cli-'));
    const outBase = path.join(tmp, 'report');
    const res = runCli(['--root', fixture, '--out', outBase, '--onlyOrphans']);
    expect(res.status).toBe(0);
    const data = res.json as any[];
    const a = data.find(r => r.node.endsWith('A.ts'))!;
    expect(a.symbols.imports[0]).toEqual(expect.objectContaining({ source: './B', target: expect.any(Object) }));
    expect(data).toMatchSnapshot();
  });
});


