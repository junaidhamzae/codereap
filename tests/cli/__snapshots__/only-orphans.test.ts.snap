// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CLI --onlyOrphans with target enrichment includes symbols.imports with target info for resolvable imports 1`] = `
[
  {
    "exists": true,
    "in-degree": 0,
    "node": "A.ts",
    "orphan": true,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
      "imports": [
        {
          "imported": {
            "default": false,
            "named": [],
            "namespace": false,
          },
          "kind": "esm",
          "resolved": "B.ts",
          "source": "./B",
          "target": {
            "exports": {
              "hasDefault": false,
              "named": [
                "b",
              ],
              "reExports": [],
            },
            "node": "B.ts",
          },
        },
      ],
    },
  },
  {
    "exists": true,
    "in-degree": 1,
    "node": "B.ts",
    "orphan": true,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "b",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
      "imports": [],
    },
  },
]
`;
